trieste/bayesian_optimizer.py

```diff
@@ -99,14 +126,14 @@ class OptimizationResult(Generic[StateType]):
     exception.
     """
 
-    history: list[Record[StateType]]
+    history: list[FrozenRecord[StateType]]
```

Comment:
could there be any issues like write protections? we should perhaps test it and raise errors if filesystem is not accessible? 

not sure if there is a merit to support both in memory and on disk and leave to the user to choose one?

Reply:
(Yes, this definitely needs error handling. And I think it probably _does_ make sense to allow both in-memory and on-disk tracking, though it will make the interfaces more complicated.)